exercises/practice/rest-api/.meta/Example.roc

```diff
@@ -0,0 +1,214 @@
+        |> List.walkTry (Ok EQ) \_state, cmp ->
+            when cmp is
+                EQ -> Ok EQ
+                res -> Err res
+    when result is
+        Ok _cmp -> Num.compare (List.len b1) (List.len b2)
+        Err res -> res
+
+## Convert the owes and owedBy dictionaries to JSON
+## Note: this will be simpler once Roc supports Encoding & Decoding Dict
+oweDictToJson : Dict Str F64 -> Str
+oweDictToJson = \dict ->
+    dictContent =
+        dict
+        |> Dict.toList
+        |> List.sortWith \(name1, _amount1), (name2, _amount2) -> compareStrings name1 name2
+        |> List.map \(name, amount) -> "$(name |> stringToJson): $(amount |> Num.toStr)"
+        |> Str.joinWith ","
+    "{$(dictContent)}"
+
+stringToJson : Str -> Str
+stringToJson = \string ->
+    result =
+        Encode.toBytes string Json.utf8
+        |> Str.fromUtf8
+    when result is
+        Ok json -> json
+        Err (BadUtf8 _ _) -> crash "Unreachable: encoding a string to JSON should never fail"
+
+## Convert a user to a JSON representation
+## Note: this will be simpler once Roc supports Encoding & Decoding Dict
+userToJson : User -> Str
+userToJson = \user ->
+    """
+    {
+        \"balance\" : $(user.balance |> Num.toStr),
+        \"name\" : $(user.name |> stringToJson),
+        \"owed_by\" : $(user.owedBy |> oweDictToJson),
+        \"owes\" : $(user.owes |> oweDictToJson)
+    }
+    """
+
+## Convert a list of users to a JSON representation
+## Note: this will be simpler once Roc supports Encoding & Decoding Dict
+usersToJson : List User -> Str
+usersToJson = \users ->
+    sortedUsers = users |> List.sortWith \u1, u2 -> compareStrings u1.name u2.name
+    # Encode.toBytes { users: sortedUsers } Json.utf8 |> Str.fromUtf8
+    listContent =
+        sortedUsers
+        |> List.map \user -> userToJson user
+        |> Str.joinWith ",\n"
+    "{\"users\": [$(listContent)]}"
+
+## Parse the requested user name or list of user names, and load them from
+## the (mock) database
+getUsers : Database, Str -> Result Str [InvalidJson]
+getUsers = \database, payload ->
+    if payload == "" then
+        database.users |> usersToJson |> Ok
+    else
+        names = getUserNames? payload
+        database.users
+        |> List.keepIf \user ->
+            names |> List.contains user.name
+        |> usersToJson
+        |> Ok
+
+## Parse the requested user names from JSON
+getUserNames : Str -> Result (List Str) [InvalidJson]
+getUserNames = \payload ->
+    when parseJsonUser payload is
+        Ok userRecord -> Ok [userRecord.user]
+        Err InvalidJson ->
+            usersRecord = parseJsonUsers? payload
+            Ok usersRecord.users
+
+## Parse a JSON { user: name } object
+parseJsonUser : Str -> Result { user : Str } [InvalidJson]
+parseJsonUser = \payload ->
+    bytes = payload |> Str.toUtf8
+    maybeUser : Decode.DecodeResult { user : Str }
+    maybeUser = Decode.fromBytesPartial bytes Json.utf8
```

Comment:
```suggestion
    maybeUser = Decode.fromBytesPartial bytes Json.utf8
```
I suggest removing this type annotation as the inferred types are enough to determine how to decode.

You could also do something like this, but perhaps it is better to introduce some intermediate defs :slightly_smiling_face: 
```roc
parseJsonUser = \payload ->
    payload
    |> Str.toUtf8
    |> Decode.fromBytesPartial Json.utf8
    |> .result
    |> Result.mapErr \_ -> InvalidJson
```